apache2/waf_logging/waf_log_util.cc

```diff
@@ -167,12 +220,34 @@ void free_json(char* str) {
     std::free(str);
 }
 
-void init_appgw_rules_id_hash() {
-    std::ifstream ifs("/etc/nginx/modsec/disableableRules.conf");
+static rule_ids_vector read_disableable_rule_ids(const std::string& filename) {
+    rule_ids_vector rule_ids;
+    std::ifstream ifs(filename);
     if(ifs) {
         std::copy(std::istream_iterator<int>(ifs),
                 std::istream_iterator<int>(),
-                std::back_inserter(default_disableable_rule_ids));
-        std::sort(default_disableable_rule_ids.begin(), default_disableable_rule_ids.end());
+                std::back_inserter(rule_ids));
+        std::sort(std::begin(rule_ids), std::end(rule_ids));
+    }
+
+    return rule_ids;
+}
+
+void init_appgw_rules_id_hash() {
+    std::map<rule_set_info, std::string> pathsMap {
```

Comment:
Can we not hardcode this map? I am trying to avoid too many changes when a new version gets added.

Reply:
I don't really see how we can not hardcode it. We could instead read it from yet another file that the tenant would write, but honestly I'm not sure this gives us any edge.
For any new type/version of a ruleset we will have to make some changes to the logging code to properly support it.

Reply:
Discussed offline.